controller/dataplane/owned_deployment.go

```diff
@@ -271,6 +280,43 @@ func listOrReduceDataPlaneDeployments(
 	return false, &deployments[0], nil
 }
 
+func podTemplateSpecHasRestartAnnotation(template *corev1.PodTemplateSpec) (string, bool) {
+	if template == nil || template.Annotations == nil {
+		return "", false
+	}
+	return template.Annotations[restartAnnotationKey], template.Annotations[restartAnnotationKey] != ""
```

Comment:
We should be able to do:

```suggestion
	return template.Annotations[restartAnnotationKey]
```

right?

Reply:
No, we can't.

When we put it in the return, it can only return one value, which does not match the function's signature.
You can test here: https://go.dev/play/p/DYwwm-re340

Reply:
TIL. Thanks for checking this.

So it seems we can still do the following to save on the double lookup:

```suggestion
	v, ok := template.Annotations[restartAnnotationKey]
	return v, ok
```

right?

Reply:
@pmalek  I don't think so. Because this function is meant to check for the existence of a restart annotation. If its value is empty `""`, that is not expected,  `v, ok := template.Annotations[restartAnnotationKey]` ok will be `true`.

This will lead to logical anomalies later on (however, I have also added relevant error handling for it; in this case, I will assume it is a restart operation).

Reply:
I see your concern I'm ok with adding this type of checks but OTOH I believe this is defensive programming. I'm just curious: is `""` possible under normal operating conditions?

Reply:
If we insist on leaving the check in, then this piece of code can become:

```suggestion
	v, ok := template.Annotations[restartAnnotationKey]
	return v, ok && v != ""
```